File: solutions/erlang/resistor-color/1/src/resistor_color.erl
-module(resistor_color).

-export([colors/0, color_code/1]).

colors() ->
  [black, brown, red, orange, yellow,
   green, blue, violet, grey, white].

color_code(C) -> find_index(C, colors(), 0).

find_index(E, [E|_], Acc) -> Acc;
find_index(E, [_|T], Acc) -> find_index(E, T, Acc + 1).
    

File: solutions/lua/resistor-color/1/resistor-color.lua
return {
  color_code = function(color)
    if color == 'black' then return 0;
    elseif color == 'brown' then return 1;
    elseif color == 'red' then return 2;
    elseif color == 'orange' then return 3;
    elseif color == 'yellow' then return 4;
    elseif color == 'green' then return 5;
    elseif color == 'blue' then return 6;
    elseif color == 'violet' then return 7;
    elseif color == 'grey' then return 8;
    elseif color == 'white' then return 9;
    else return 42;
    end
  end
}
